api/src/main/java/io/cloudevents/impl/CloudEventImpl.java

```diff
@@ -71,53 +109,6 @@ public CloudEvent toV1() {
         );
     }
 
-    // Message impl
-
-    public BinaryMessage asBinaryMessage() {
-        return this;
-    }
-
-    public StructuredMessage asStructuredMessage(EventFormat format) {
-        CloudEvent ev = this;
-        // TODO This sucks, will improve later
-        return new StructuredMessage() {
-            @Override
-            public <T> T visit(StructuredMessageVisitor<T> visitor) throws MessageVisitException, IllegalStateException {
-                return visitor.setEvent(format, format.serialize(ev));
-            }
-        };
-    }
-
-    @Override
-    public void visitExtensions(BinaryMessageExtensionsVisitor visitor) throws MessageVisitException {
```

Comment:
One of the core pattern of message/event exchanges is their immutability since events represent moment in history and this can not be re-written. So giving user reference to a mutable map is very dangerous